app/services/my_anime_list_sync_service.rb

```diff
@@ -0,0 +1,166 @@
+class MyAnimeListSyncService
+  ATARASHII_API_HOST = 'https://hbv3-mal-api.herokuapp.com/2.1/'.freeze
+  MINE = '?mine=1'.freeze
+
+  attr_reader :library_entry, :method
+
+  def initialize(library_entry, method)
+    @library_entry = library_entry
+    @method = method
+  end
+
+  def execute_method
+    case method
+    when 'delete'
+      delete("#{media_type}list/#{media_type}/#{mal_media_id}", linked_profile)
+    when 'create/update'
+      # find the anime or manga
+      # it will raise an error if it fails the http request
+      response = get("#{media_type}/#{mal_media_id}#{MINE}", linked_profile)
+
+      if media_type == 'anime'
+        if response['watched_status']
+          put("animelist/anime/#{mal_media_id}", linked_profile,
+            status: format_status(library_entry.status),
+            episodes: library_entry.progress,
+            score: format_score(library_entry.rating),
+            rewatch_count: library_entry.reconsume_count)
+        else
+          post('animelist/anime', linked_profile,
+            anime_id: mal_media_id,
+            status: format_status(library_entry.status),
+            episodes: library_entry.progress,
+            score: format_score(library_entry.rating))
+        end
+      else
+        # manga
+        if response['read_status']
```

Comment:
Instead of doing this:

```ruby
if foo
  # do stuff
else
  if bar
    # do other stuff
  else
    # do other other stuff
  end
end
```

Do this

```ruby
if foo
  # do stuff
elsif bar
  # do other stuff
else
  # do other other stuff
end
```